models.js

```diff
@@ -46,10 +45,37 @@ dbModel.prototype = {
   save: function(objData, options) {
     // MORETODO: returns a promise at the moment
     // mostly NOT DONE. options only has {conflict: 'update'} possibility
+    var self = this
     if (Array.isArray(objData)) {
-      return this.kninky.k.batchinsert(this.tableName, objData)
+      console.log('SAVE', objData.length, objData[0], options)
+      return this.kninky.k.batchInsert(this.tableName, objData, 100)
     } else {
+      console.log('SAVE', objData, options)
+      if (this.kninky.defaultsUnsupported) {
+        for (var a in this.fields) {
+          var f = this.fields[a]
+          if (typeof objData[a] == 'undefined'
+              && typeof f.defaultVal != 'undefined') {
+            objData[a] = f.defaultVal
+          }
+        }
+        console.log('SAVE w/defaults', objData)
+      }
+      if (options.conflict == 'update' && objData.id) {
+        //STARTHERE: look for val, if so, then send update
+        //need to turn the insert into a function which waits on result
+        return this.kninky.k.table(this.tableName)
```

Comment:
lol --> replace all 'r.' with 'k.' --> done!
thanks ok, coming together for me!